packages/flame/lib/src/components/input/button_component.dart

```diff
@@ -45,17 +45,18 @@ class ButtonComponent extends PositionComponent with Tappable {
       button != null,
       'The button has to either be passed in as an argument or set in onLoad',
     );
-    final idleButton = button;
-    if (idleButton != null && !contains(idleButton)) {
-      add(idleButton);
+    if (!contains(button!)) {
+      add(button!);
     }
   }
 
   @override
   @mustCallSuper
   bool onTapDown(TapDownInfo info) {
-    button?.removeFromParent();
-    buttonDown?.changeParent(this);
+    if (buttonDown != null) {
+      button!.parent = null;
```

Comment:
I think it looks nicer to avoid setting the parent to null explicitly, and only let the method handle the remove method handle that.

Reply:
ok